routers/web/repo/editor.go

```diff
@@ -45,6 +46,41 @@ func canCreateBasePullRequest(ctx *context.Context) bool {
 	return baseRepo != nil && baseRepo.UnitEnabled(ctx, unit.TypePullRequests)
 }
 
+// redirectToFork redirects to the user's fork when editing is not possible
+func redirectToFork(ctx *context.Context) bool {
+	if ctx.Repo.CanWriteToBranch(ctx, ctx.Doer, ctx.Repo.BranchName) {
+		return false
+	}
+
+	repo := ctx.Repo.Repository
+	userAndOrgForks, err := repo_model.GetForksByUserAndOrgs(ctx, ctx.Doer, repo)
+	if err != nil {
+		ctx.ServerError("GetForksByUserAndOrgs", err)
+		return true
+	}
+
+	// When already on a repo the user owns, don't redirect further even if
+	// writing to the branch is not possible.
+	for _, forkRepo := range userAndOrgForks {
+		if forkRepo.ID == repo.ID {
```

Comment:
I don't quite understand the logic. All repositories on `userAndOrgForks` should be forks of `repo`, looks like it's impossible `forkRepo.ID == repo.ID`.